src/providers/oracle/qgsoracleprovider.h

```diff
@@ -265,9 +261,10 @@ class QgsOracleProvider final: public QgsVectorDataProvider
     QString mPrimaryKeyDefault;
 
     /**
-     * List of always generated key attributes
+     * For each attributes, true if the attribute is always generated (virtual column
+     * and identity always generated columns)
```

Comment:
something missing?

Reply:
not really, what do you think is missing ?

Reply:
I didn't understand the part in brackets: are identity and virtual column generated columns themselves? or do they generate columns?
Now that I (re)read, since it applies to attributes, do we also need "for each attributes"?
PS: Keep in mind that I barely understand half of what the PR is doing 🤔 so sorry for any noise.

Reply:
> identity and virtual column generated columns themselves? or do they generate columns?

When defining a table attribute In Oracle, you can define
- a *virtual* column: The column is generated based upon an expression
-  an *identity always generated* column: Its  an auto generated integer (same as *serial* in PostGres) which cannot be modified (*always generated*)

So that's two ways to have generated, not editable column. That's the meaning of the part inside the brackets.

> Now that I (re)read, since it applies to attributes, do we also need "for each attributes"?

I'd say yes, it's not completely obvious in this part of the code that the list contains a value for each attributes. 

Thanks for your review!

Reply:
> an _identity always generated column_: Its an auto generated integer (same as serial in PostGres) which cannot be modified (always generated)

Ah... OK! the column type name is "identity always generated". I thought "always generated" was another part of the sentence. As I said... 😄